ocaml/testsuite/tests/pattern-guards/test.ml

```diff
@@ -8,16 +8,30 @@ let basic_usage ~f ~default x =
   match x with 
     | Some x when f x match Some y -> y
     | _ -> default
-[%%expect{|
-Uncaught exception: Failure("guard pattern translation unimplemented")
+;;
 
+let f x = if x > 0 then Some (x - 1) else None;;
+
+basic_usage ~f ~default:~-1 (Some 5);;
+basic_usage ~f ~default:~-1 (Some 0);;
+basic_usage ~f ~default:~-1 (Some ~-5);;
+basic_usage ~f ~default:~-1 None;;
+[%%expect{|
```

Comment:
Writing more `%%expect` blocks has the advantage that the output is more clearly associated with the code that generated it. Could you consider, here and throughout, whether writing more `%%expect` blocks would make the code easier to read?